src/controller.cpp

```diff
@@ -0,0 +1,68 @@
+#include "controller.hpp"
+
+MotorController::Controller::Controller() : dcInterface(), stepperInterface() {
+}
+
+MotorController::Controller::INTERFACE MotorController::Controller::getSelectedInterface() const {
+    // Getting while interface is nullptr
+    if (selectedInterface == nullptr) {
+        hwlib::cout << "Error:: Trying to get an nullptr interface (controller.cpp, line: " << __LINE__ << ")" << hwlib::endl;
+        // Better not continue
+        while (1) {

```

Comment:
This loop would prevent error handling, (e.g. check if an interface has been selected, otherwise do {some action}), its advised to add an enum item named "Invalid" and return that instead.